core/validation/utils.py

```diff
@@ -12,3 +17,88 @@ def remove_duplicate_indexes(df: pd.DataFrame):
     :return: the DataFrame without duplicate indexes
     """
     return df[~df.index.duplicated(keep="first")]
+
+
+def get_cell_indexes_for_outcomes(failed_row: pd.Series, index: int) -> str:
+    """
+    Constructs cell indexes for outcomes based on the provided failed row and index.
+
+    The function determines whether the error occurred in footfall outcomes (starting from row 60)
+    or non-footfall outcomes. If the index is greater than or equal to 60, it calculates the cell
+    index for footfall outcomes based on the provided 'Start_Date'. If the index is less than 60,
+    it calculates the cell index for non-footfall outcomes.
+
+    For footfall outcomes, the cell index is determined by adding a row index gap calculated from
+    the start year of the UK financial year as each year represents an additional 5 rows in the
+    original spreadsheet from the row where the entry for a given footfall outcome's data begins.
+
+    :param failed_row: A pandas Series representing a row where an error has occurred.
+    :param index: The index indicating the position of the error in the data.
+    :return: A string containing the constructed cell index for outcomes.
+    """
+    start_date = failed_row["Start_Date"]
+
+    # footfall outcomes starts from row 60
+    if index >= 60:
+        financial_year = get_uk_financial_year_start(start_date)
+        # row for 'Amount' column is end number of start year of financial year * 5 + 'Footfall Indicator' index
+        row_index_gap = int(financial_year[-1]) * 5
+        index = str(int(index) + row_index_gap)
+        cell_index = footfall_outcome_amount_to_col(start_date).format(i=index)
+    else:
+        index = str(index)
+        cell_index = non_footfall_outcome_amount_to_col(start_date).format(i=index)
+
+    return cell_index
+
+
+def get_uk_financial_year_start(start_date: str) -> str:
+    """
+    Gets the start year of the UK financial year based on the provided start date.
+
+    :param start_date: A string representing the start date in the format '%Y-%m-%d %H:%M:%S'.
+    :return: A string representing the start year of the UK financial year.
+    """
+    start_date = datetime.strptime(start_date, "%Y-%m-%d %H:%M:%S")
+
+    financial_year_start = datetime(start_date.year, 4, 1)
+    if start_date < financial_year_start:
+        financial_year_start = datetime(start_date.year - 1, 4, 1)
+
+    return f"{str(financial_year_start.year)}"
```

Comment:
we pass around datetimes instead of strings this function can just be this I think?
```
financial_year = date.year if date.month >= 4 else date.year - 1
return str(financial_year)
```